iModelCore/ECPresentation/Tests/NonPublished/Integration/Content/ContentIntegrationTests.cpp

```diff
@@ -953,6 +953,194 @@ TEST_F(RulesDrivenECPresentationManagerContentTests, DescriptorOverride_FiltersB
+    <ECEntityClass typeName="A">
+        <ECProperty propertyName="GuidProp" typeName="binary" extendedTypeName="BeGuid" />
+    </ECEntityClass>
+)*");
+TEST_F(RulesDrivenECPresentationManagerContentTests, Guid_AppliesExtendedDataRule)
+    {
+    ECClassCP classA = GetClass("A");
+    BeGuid instanceGuid1;
+    instanceGuid1.FromString("182238d2-e836-4640-9b40-38be6ca49623");
+    IECInstancePtr instance1 = RulesEngineTestHelpers::InsertInstance(s_project->GetECDb(), *classA, [instanceGuid1](IECInstanceR instance) {
+        instance.SetValue("GuidProp", ECValue((Byte*)&instanceGuid1, sizeof(BeGuid)));
+        });
+
+    KeySetPtr input = KeySet::Create(bvector<IECInstancePtr>{instance1});
+
+    // create the rule set
+    PresentationRuleSetPtr rules = PresentationRuleSet::CreateInstance(BeTest::GetNameOfCurrentTest());
+    m_locater->AddRuleSet(*rules);
+    ContentRuleP contentRule = new ContentRule("", 1, false);
+    ContentInstancesOfSpecificClassesSpecification* spec = new ContentInstancesOfSpecificClassesSpecification(1, "", classA->GetFullName(), false, false);
+    contentRule->AddSpecification(*spec);
+    rules->AddPresentationRule(*contentRule);
+
+    ExtendedDataRule* ex1 = new ExtendedDataRule();
+    ex1->AddItem("stringified_guid", "GuidToStr(this.GuidProp)");
+    ex1->AddItem("not_stringified_guid", "this.GuidProp");
+    rules->AddPresentationRule(*ex1);
+
+    // validate descriptor
+    ContentDescriptorCPtr descriptor = GetValidatedResponse(m_manager->GetContentDescriptor(AsyncContentDescriptorRequestParams::Create(s_project->GetECDb(), rules->GetRuleSetId(), RulesetVariables(), nullptr, 0, *input)));
+    ASSERT_TRUE(descriptor.IsValid());
+
+    // validate content
+    ContentCPtr content = GetVerifiedContent(*descriptor);
+    ASSERT_TRUE(content.IsValid());
+    DataContainer<ContentSetItemCPtr> contentSet = content->GetContentSet();
+    ASSERT_EQ(1, contentSet.GetSize());
+
+    RapidJsonAccessor extendedData1 = contentSet[0]->GetUsersExtendedData();
+    ASSERT_TRUE(extendedData1.GetJson().IsObject());
+    ASSERT_EQ(2, extendedData1.GetJson().MemberCount());
+
+    ASSERT_TRUE(extendedData1.GetJson().HasMember("stringified_guid"));
+    EXPECT_STREQ(instanceGuid1.ToString().c_str(), extendedData1.GetJson()["stringified_guid"].GetString());
+
+    ASSERT_TRUE(extendedData1.GetJson().HasMember("not_stringified_guid"));
+    ASSERT_TRUE(extendedData1.GetJson()["not_stringified_guid"].IsNull());
+    }
+
+/*---------------------------------------------------------------------------------**//**
+* @bsitest
++---------------+---------------+---------------+---------------+---------------+------*/
+DEFINE_SCHEMA(Guid_FilteringUsingSpecificationsFilter, R"*(
+    <ECEntityClass typeName="A">
+        <ECProperty propertyName="GuidProp" typeName="binary" extendedTypeName="BeGuid" />
+    </ECEntityClass>
+)*");
+TEST_F(RulesDrivenECPresentationManagerContentTests, Guid_FilteringUsingSpecificationsFilter)

```

Comment:
I'd mode the test next to `ContentInstancesOfSpecificClasses_InstanceFilter` and rename to `ContentInstancesOfSpecificClasses_InstanceFilter_UsingGuidProperty`